src/devices/nm-lldp-listener.c

```diff
@@ -268,6 +283,21 @@ _lldp_attr_set_uint32 (LldpAttrData *pdata, LldpAttrId attr_id, guint32 v_uint32
 	pdata->v_uint32 = v_uint32;
 }
 
+static void
+_lldp_attr_add_vardict (LldpAttrData *pdata, LldpAttrId attr_id, GVariant *variant)
```

Comment:
it's wrong that the function requires a non-floating variant and takes ownership of it. API should either accept a floating reference (and consume it) or take an additional reference on the input. That means, call `g_variant_ref_sink(variant)` and ensure that the callers get adjusted (to not sink the reference themselves and leak it).